iroh/src/node/builder.rs

```diff
@@ -715,6 +690,229 @@ where
     }
 }
 
+/// A node that is initialized but not yet spawned.
+///
+/// This is returned from [`Builder::build`] and may be used to register custom protocols with
+/// [`Self::accept`]. It provides access to the services which are already started, the node's
+/// endpoint and a client to the node.
+///
+/// Note that the client returned from [`Self::client`] can only be used after spawning the node,
+/// until then all RPC calls will time out.
+#[derive(derive_more::Debug)]
+pub struct UnspawnedNode<D, E> {
```

Comment:
Maybe call this CustomProtocolBuilder since you will use it only when you want to register custom protocols? Naming 🤷

Reply:
Went with `ProtocolBuilder` for now. Open to changing further :)